index.js

```diff
@@ -10,8 +10,69 @@ const COLORS = 'https://nt-cdn.s3.amazonaws.com/colors.json';
  * @param compHex filter for complementary color hex code
  * @returns Promise
  */
-const fetchColors = ({ name, hex, compName, compHex }) => {
-  throw Error('Not implemented');
+
+const fetchColors = async ({ name, hex, compName, compHex }) => {
+  // use fetch api to retrieve json entries
+  try {
+    // set variable "response" equal to the response object returned from fetch
+    const response = await fetch(COLORS);
+    // if something foes wrong, put up this error message
+    if (!response.ok) {
+      throw new Error("aiya");
+    }
+    // set variable "data" equal to the response body in json format
+    const data = await response.json();
+    
+    // create empty variable to set answer equal to
+    let result;
+
+    // now use switch statements to fork the function if certain inputs are detected
+    switch (true) {
```

Comment:
This is an interesting way of writing it. It reads quite well actually. A bit verbose and not the biggest fan of the `switch` construct in general (I always forget the `break`!) but this is purely personal.

Reply:
A simple `if/else if` would do here. Nothing wrong with the `switch` though, just hadn't seen it used like that.